ext/opcache/jit/zend_jit_trace.c

```diff
@@ -7602,6 +7602,25 @@ static void zend_jit_blacklist_root_trace(const zend_op *opline, size_t offset)
 	zend_shared_alloc_unlock();
 }
 
+ZEND_EXT_API void zend_jit_blacklist_function(zend_op_array *op_array) {
+	zend_jit_op_array_trace_extension *jit_extension = (zend_jit_op_array_trace_extension *)ZEND_FUNC_INFO(op_array);
+	if (!jit_extension) {
+		return;
+	}
+
+	zend_jit_stop_persistent_op_array(op_array);
+
+	// First not-skipped op
+	zend_op *opline = op_array->opcodes;
+	if (!(op_array->fn_flags & ZEND_ACC_HAS_TYPE_HINTS)) {
+		while (opline->opcode == ZEND_RECV || opline->opcode == ZEND_RECV_INIT) {
+			opline++;
+		}
+	}
+
+	zend_jit_blacklist_root_trace(opline, jit_extension->offset);
```

Comment:
Reset ``ZEND_FUNC_JIT_ON_HOT_TRACE`` in ``jit_extension->func_info.flags`` instead.